seq2science/rules/peak_counts.smk

```diff
@@ -301,3 +268,43 @@ rule mean_center:
             f"# The number of reads under each peak, mean centered after log1p {wildcards.base} and {wildcards.normalisation} normalization\n"
             + cov_mc.to_csv(index=True, header=True, sep="\t")
         )
+
+
+def get_all_narrowpeaks(wildcards):
+    return [f"{config['result_dir']}/{{peak_caller}}/{{assembly}}-{replicate}_peaks.narrowPeak"
+        for replicate in breps[breps['assembly'] == wildcards.assembly].index]
+
+
+rule onehot_peaks:
+    """
+    Get onehot encodings of which peaks are found in which samples
+    """
+    input:
+        narrowpeaks=get_all_narrowpeaks,
+        combinedpeaks=rules.combine_peaks.output
+    output:
+        real=expand("{result_dir}/{{peak_caller}}/{{assembly}}_onehotpeaks.tsv", **config),
+        tmp=temp(expand("{result_dir}/{{peak_caller}}/{{assembly}}_onehotpeaks.tsv.tmp", **config))
+    conda:
+        "../envs/bedtools.yaml"
+    log:
+        expand("{log_dir}/onehot_peaks/{{assembly}}-{{peak_caller}}.log", **config),
+    benchmark:
+        expand("{benchmark_dir}/onehot_peaks/{{assembly}}-{{peak_caller}}.benchmark.txt", **config)[0]
+    params:
+        peak_count=lambda wildcards, input: len(input.narrowpeaks)
+    shell:
+        """
+        awk '{{print $1":"$2"-"$3}}' {input.combinedpeaks} > {output.real} 2> {log}
+
+        for brep in {input.narrowpeaks}
+        do
+            bedtools sort -i $brep 2> {log} |
+            bedtools intersect -a {input.combinedpeaks} -b stdin -c 2> {log} |
+            awk '{{print $4}}' 2> {log} |
+            paste {output.real} - > {output.tmp}
+            mv {output.tmp} {output.real}
```

Comment:
If I understand correctly, you are trying to append each `brep` to the output file. I think you could do that more simply with `awk '{{print $4}}' >> {output.real} 2> {log}` (`>>` creates a file if missing, and appends if existing)

Furthermore, you could start with the echo below. That way you can omit the use of `output.tmp`. If an error occurs, snakemake will delete the file anyway, so no need to add the comment only after the loop.

Reply:
It adds columns, not rows, you end up with a file like this:

```
    sample1    sample2    sample3
peak1    0    1    0
peak2    1    1    1
peak3    0    0    1

```

Reply:
aah. Thanks!